### ISSUE TITLE
Dependabot PRs don't pass CI

### ISSUE THREAD

**Issue by mmarchetti** [posit-dev/publisher#1174]
Dependabot PRs (e.g. https://github.com/posit-dev/publisher/pull/1156) are failing CI on the BATS tests becasue they lack a token to do the Connect checkout.

**Comment by dotNomad** [comment]
Help FAQ about dependabot workflow permissions here: https://github.com/dependabot/dependabot-core/issues/3253#issuecomment-852541544

One thing to note is we can set secrets for Dependabot specifically in our repo settings, but its worth noting that malicious packages could try to exfiltrate credentials: https://eslint.org/blog/2018/07/postmortem-for-malicious-package-publishes/

The `pull_request_target` method in the FAQ seems like one of the best options 🤔

**Comment by dotNomad** [comment]
We could provide a Connect license to dependabot through the newer repo secrets settings. That would fix:

> Error when evaluating 'secrets'. .github/workflows/pull-request.yaml (Line: 22, Col: 11): Secret CONNECT_LICENSE is required, but not provided while calling.

**Comment by jonkeane** [comment]
We also could consider [disabling running](https://docs.github.com/en/code-security/dependabot/troubleshooting-dependabot/troubleshooting-dependabot-on-github-actions#troubleshooting-failures-when-dependabot-triggers-existing-workflows) the integration (or more accurately the tests that require credentials) for dependabot PRs. There is a pretty low liklihood that only those would fail integration tests, and if they did after we merge, the revert is incredibly simple.

**Comment by dotNomad** [comment]
> I _think_ we can safely skip the following in `pull-request.yaml` for Dependabot PR as those will execute once it's merged to main via `main.yaml`. [@dotNomad](https://github.com/dotNomad) does that sound reasonable? Could we skip these for ALL PRs?
> 
> * archive.yaml
> * upload.yaml

Having the built artifact VSIX uploaded from `archive.yaml` is helpful for PRs. I'm not sure why that one would fail for dependabot though?

Skipping `upload.yaml` makes sense. Those look like they are pushing up artifacts to AWS which I believe we can stop doing since we rely on the marketplaces and GitHub releases now.

> We can get e2e running by giving Dependabot CONNECT_LICENSE. Do we have any concerns with this?

Nothing immediately comes to mind other that isn't also an issue for local development - like compromised `npm` packages.

**Comment by dotNomad** [comment]
> I don't have access to whatever license we're already using for e2e and I don't see any mention of specific class of license required. Do you have this information?

Yeah once it is uploaded to GitHub I don't think anyone has access to it. I didn't set it up so I'm not sure what class of license is used / needed.
